test/e2e/ratelimit/ratelimit_universal.go

```diff
@@ -175,4 +191,41 @@ conf:
 		// verify determinism by running it once again with shorter timeout
 		Eventually(verifyRateLimit("web", 5), "30s", "1s").Should(Equal(1))
 	})
+
+	It("should limit echo server as external service", func() {
+		externalService := `
+type: ExternalService
+mesh: default
+name: external-service
+tags:
+  kuma.io/service: external-service
+  kuma.io/protocol: http
+networking:
+  address: "kuma-3_externalservice-http-server:81"
+`
+		specificRateLimitPolicy := `
+type: RateLimit
+mesh: default
+name: rate-limit-demo-client
+sources:
+- match:
+    kuma.io/service: "demo-client"
+destinations:
+- match:
+    kuma.io/service: "external-service"
+conf:
+  http:
+    requests: 4
+    interval: 10s
+`
```

Comment:
Add here one more tag and see if it still works
```
---
type: ExternalService
mesh: default
name: external-service
tags:
  kuma.io/service: external-service
  kuma.io/protocol: HTTP
  version: v1
networking:
  address: "kuma-3_externalservice-http-server:81"
---
type: RateLimit
mesh: default
name: rate-limit-demo-client
sources:
- match:
    kuma.io/service: "demo-client"
destinations:
- match:
    kuma.io/service: "external-service"
    version: v1
conf:
  http:
    requests: 4
    interval: 10s
```

I think it won't but I might be wrong

Reply:
I tested it myself and see that RateLimit is not set. Let's pair to figure out what is wrong with either setup

Reply:
Was able to reproduce problem. Working on solution..